tests/s3/copy_object/test_parallel_copy_object.py

```diff
@@ -0,0 +1,402 @@
+
+        LOGGER.info("Upload object1 to source bucket1")
+        self.create_put_object("src-bucket", "obj1", 100)
+
+        self.parallel_copy_and_put_object(("src-bucket", "obj1", "src-bucket", "obj2"),
+                                          ("src-bucket", "obj2", self.file_path))
+
+        LOGGER.info("Upload multipart object obj3 to src-bucket")
+        self.s3mp_obj.simple_multipart_upload("src-bucket", "obj3", 1024, self.file_path, 4)
+
+        self.parallel_copy_and_put_object(("src-bucket", "obj3", "src-bucket", "obj4"),
+                                          ("src-bucket", "obj4", self.file_path))
+
+        LOGGER.info("All objects should be listed in relevant buckets")
+        self.objects_in_buckets(bucket="src-bucket", objects=["obj1", "obj2", "obj3", "obj4"])
+
+        LOGGER.info("ENDED: Test Parallel put and copy on destination object "
+                    "(simple and multipart source objects)")
+
+    @pytest.mark.tags("TEST-44807")
+    @CTFailOn(error_handler)
+    def test_44807(self):
+        """Test Parallel copy and overwrite of same source object"""
+        LOGGER.info("STARTED: Test Parallel copy and overwrite of same source object "
+                    "(simple and multipart source object)")
+
+        LOGGER.info("Upload object1 to source bucket1")
+        self.create_put_object("src-bucket", "obj1", 100)
+
+        self.parallel_copy_and_put_object(("src-bucket", "obj1", "src-bucket", "obj2"),
+                                          ("src-bucket", "obj2", self.file_path))
+
+        LOGGER.info("Upload multipart object obj3 to src-bucket")
+        self.s3mp_obj.simple_multipart_upload("src-bucket", "obj3", 1024, self.file_path, 4)
+
+        self.parallel_copy_and_put_object(("src-bucket", "obj3", "src-bucket", "obj4"),
+                                          ("src-bucket", "obj3", self.file_path))
+
+        LOGGER.info("All objects should be listed in relevant buckets")
+        self.objects_in_buckets(bucket="src-bucket", objects=["obj1", "obj2", "obj3", "obj4"])
+
+        LOGGER.info("ENDED: Test Parallel copy and overwrite of same source object "
+                    "(simple and multipart source objects)")
+
+    @pytest.mark.tags("TEST-44792")
+    @CTFailOn(error_handler)
+    def test_44792(self):
+        """Test Parallel Self copy and put object in same bucket"""
+        LOGGER.info("STARTED: Test Parallel Self copy and put object in same bucket "
+                    "(simple and multipart source object)")
+
+        LOGGER.info("Upload object1 to source bucket1")
+        self.create_put_object("src-bucket", "obj1", 100)
+
+        self.parallel_copy_and_put_object(
+            ("src-bucket", "obj1", "src-bucket", "obj1", "InvalidRequest"),
+            ("src-bucket", "obj2", self.file_path))
+
+        LOGGER.info("Upload multipart object obj3 to src-bucket")
+        self.s3mp_obj.simple_multipart_upload("src-bucket", "obj3", 1024, self.file_path, 4)
```

Comment:
lets make it 3 gb object and rename it to mp-obj1 like above tests. I will modify name in tds and test